public/app/percona/add-instance/components/AddInstance/AddInstance.tsx

```diff
@@ -43,7 +46,17 @@ export const AddInstance: FC<AddInstanceProps> = ({ onSelectInstanceType, showAz
     [showAzure]
   );
 
-  const selectInstanceType = (type: string) => () => onSelectInstanceType({ type: type as InstanceAvailableType });
+  const dispatch = useDispatch();
+  dispatch(UserFlow.startFlow(uuidv4(), 'inventory:add_instance'));
```

Comment:
If the concern "infinite update loop", there is no such a problem. UserFlow actions don't cause rerender, because reducer is not used. It is designed the way – there are no reducers available.